### ISSUE TITLE
Duplex connection support via r2pipe.

### ISSUE THREAD

**Issue by XVilka** [radare/r2pipe.rs#21]
To be able to get events from radare2 as well.

**Comment by radare** [comment]
????

**Comment by radare** [comment]
Theres no such thiny in r2pipe. So it doesnt belongs to the rust module

**Comment by XVilka** [comment]
AFAIR there were plans for r2pipe for double-way communication, which is important for radeco and related projects, imho.

**Comment by radare** [comment]
Yes but it has not been designed or implemwnted yet. So it doesnt belongs to the rust repo

**Comment by chinmaydd** [comment]
@radare do you want to design a singular duplex model which would then be implemented for all pipes?

**Comment by radare** [comment]
i would prefer to reuse the same channel but dont wanna break current r2pipe, and i dont think this is possible, in fact that stream should be about capturing stderr, anything else? maybe we can add a separate pipe.. but optional to not break compat
